src/components/Arrow/Button.tsx

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable react/destructuring-assignment */
+import * as React from "react";
+import { ArrowDirection, ArrowType, ButtonIconState } from "./Utils";
+import styles from "./Button.module.scss";
+import { createCompletedIcon, createEditIcon, createNoteIcon } from "./Icons";
+
+const renderIcon = (state: ButtonIconState, iconColor: string): JSX.Element => {
+  switch (state) {
+    case ButtonIconState.Completed:
+      return createCompletedIcon(iconColor);
+    case ButtonIconState.Notes:
+      return createNoteIcon(iconColor);
+    default:
+      return createEditIcon(iconColor);
+  }
+};
+
+export function clickArrow(): void {
+  // TODO
+}
+
+interface ArrowButtonProps {
+  arrowColor: string;
+  circleColor: string;
+  iconColor: string;
+  type: ArrowType;
+  direction: ArrowDirection;
+  buttonState: ButtonIconState;
+}
+
+export const ArrowButton: React.FC<ArrowButtonProps> = ({
+  arrowColor,
+  circleColor,
+  iconColor,
+  type,
+  direction,
+  buttonState,
+}): JSX.Element => {
+  let classNames = `${styles.button}`;
+
+  if (type !== ArrowType.Directional)
+    classNames += ` ${styles.buttonNonDirectional}`;
+  else classNames += ` ${styles.buttonDirectional}`;
+
+  switch (direction) {
+    case ArrowDirection.Up:
+      classNames += ` ${styles.buttonUp}`;
+      break;
+    case ArrowDirection.Down:
+      classNames += ` ${styles.buttonDown}`;
+      break;
+    case ArrowDirection.Left:
+      classNames += ` ${styles.buttonLeft}`;
+      break;
+    case ArrowDirection.Right:
+      classNames += ` ${styles.buttonRight}`;
+      break;
+  }
```

Comment:
This is perfectly fine, but could perhaps be cleaner as 

```suggestion
  const directionClasses = {
    [ArrowDirection.Up]: styles.buttonUp,
    [ArrowDirection.Down]: styles.buttonDown,
    [ArrowDirection.Left]: styles.buttonLeft,
    [ArrowDirection.Right]: styles.buttonRight,
  };

  classNames += ` ${directionClasses[direction]}`;
```